File: Java/ex2.java
class Carro {
    private String marca;
    private String modelo;
    private int ano;
    private int velocidade; 

    public Carro(String marca, String modelo, int ano) {
        this.marca = marca;
        this.modelo = modelo;
        this.ano = ano;
        this.velocidade = 0; 
    }

    public void acelerar(int incremento) {
        this.velocidade += incremento;
        System.out.println("Acelerando. Velocidade atual: " + this.velocidade + " km/h");
    }

    public void frear(int decremento) {
        this.velocidade -= decremento;
        if (this.velocidade < 0) {
            this.velocidade = 0; 
        }
        System.out.println("Freando. Velocidade atual: " + this.velocidade + " km/h");
    }

    public void exibirVelocidade() {
        System.out.println("Velocidade atual: " + this.velocidade + " km/h");
    }

    public void exibirDetalhes() {
        System.out.println("Marca: " + this.marca + ", Modelo: " + this.modelo + ", Ano: " + this.ano);
    }

    public static void main(String[] args) {
        
        Carro carro1 = new Carro("Toyota", "Corolla", 2020);
        Carro carro2 = new Carro("Honda", "Civic", 2021);
        Carro carro3 = new Carro("Ford", "Focus", 2019);

        
        carro1.exibirDetalhes();
        carro2.exibirDetalhes();
        carro3.exibirDetalhes();

        
        carro1.acelerar(50);
        carro1.frear(20);
        carro1.exibirVelocidade();
        carro1.frear(40); 
    }
}

File: Golang/ex2.go
package main

import (
	"fmt"
)

type Carro struct {
	Marca      string
	Modelo     string
	Ano        int
	Velocidade int
}

func (c *Carro) Acelerar(quantidade int) {
	c.Velocidade += quantidade
}

func (c *Carro) Frear(quantidade int) {
	if c.Velocidade-quantidade < 0 {
		c.Velocidade = 0
	} else {
		c.Velocidade -= quantidade
	}
}

func (c Carro) ExibirVelocidade() {
	fmt.Printf("A velocidade atual do %s %s é: %d km/h\n", c.Marca, c.Modelo, c.Velocidade)
}

func main() {

	carro1 := Carro{Marca: "Toyota", Modelo: "Corolla", Ano: 2020}
	carro2 := Carro{Marca: "Ford", Modelo: "Focus", Ano: 2019}
	carro3 := Carro{Marca: "Honda", Modelo: "Civic", Ano: 2021}

	carro1.Acelerar(50)
	carro1.ExibirVelocidade()

	carro2.Acelerar(30)
	carro2.ExibirVelocidade()

	carro3.Acelerar(70)
	carro3.ExibirVelocidade()

	carro1.Frear(20)
	carro1.ExibirVelocidade()

	carro2.Frear(50)
	carro2.ExibirVelocidade()

	carro3.Frear(100)
	carro3.ExibirVelocidade()
}
